### PR TITLE
Add N-1 Support for Subcloud Configuration

### PR DESCRIPTION
**Pull request by sharbhar**
Due to USM, the load-import functionality has been replaced with software upload. In the software upload
process for N-1 loads, deployment-manager files are no longer present in /opt/platform/software_version.

We are checking for the presence of DM deploy files in the /opt/platform/deploy/<sw_version> directory.
If these files are found, the DM files (playbook, overrides, helm-chart) will be updated in the payload.

If the files are not found in the specified directory, the process will then search for them in an alternative
location at /usr/local/share/applications/ on the system controller. If located there, these files will
update the DM files in the payload, and the user_uploaded_artifacts flag will be set to false, indicating that the files were not manually uploaded by the user.

Since the system controller playbook is updated in the payload rather than the subcloud, this means the system controller
playbook (version 24.09) will be executed within the subcloud configuration version 22.12. 

To address this, the playbook now verifies whether the host version is less than 24.09, if the subcloud and user has not
uploaded DM artifacts then helm based Deployment-manager is install.

Test-Plan:
A) PASS: Deploy AIO-SX successfully.

B) Set-up
DC-libvirt 24.09 system controller 22.12 subcloud

1. Uploaded 22.12 DM artifacts using
   dcmanager subcloud deploy upload DM-chart, overrides,
   playbook.
2. Added a 22.12 subcloud using dcmanager subcloud add.
   Validated the uploaded artifacts are picked up for
   configuration.
3. Helm based DM and DM monitor is running.
4. Deleted the DM artifacts present on the subcloud in
   /usr/local/share/applications/
5. Recong a subcloud using dcmanager subcloud deploy config,
   validated the uploaded artifacts are picked up for
   configuration.
6. Helm based DM and DM monitor is running.

1. Not uploaded 22.12 DM artifacts using
2. Added a 22.12 subcloud using dcmanager subcloud add.
   Validated the DM helm-chart and overrides are picked from
   sublcoud for configuration, and playbook from system controller.
3. Helm based DM and DM monitor is running.
4. Recong a subcloud using dcmanager subcloud deploy config,
   Validated the DM helm-chart and overrides are picked from
   sublcoud for configuration, and playbook from system controller.
5. Helm based DM and DM monitor is running.

C) Set-up
DC-libvirt 24.09 system controller 24.09 subcloud

1. Not uploaded 24.09 DM artifacts using
   dcmanager subcloud deploy upload DM-chart, overrides,
   playbook.
2. Added a 24.09 subcloud using dcmanager subcloud add.
3. System application based DM and DM monitor is running.
4. Recong a subcloud using dcmanager subcloud deploy config.
5. System application based DM and DM monitor is running.

### CODE REVIEW

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml [hunk 1]
```diff
@@ -43,11 +44,55 @@
...
+      - name: Gather all the inputs
+        set_fact:
+        vars: |
+          ansible_become_pass={{ ansible_become_pass }}
+          ansible_ssh_pass={{ ansible_become_pass }}
+          admin_password={{ admin_password }}
+          deployment_config={{ deployment_config }}
+          user_uploaded_artifacts={{ user_uploaded_artifacts }}
+
+      - name: Install Deployment Manager
```
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:85]: Error reporting is expected if N-1

**Comment by hugonicodemos** [comment]
Add the Test Plan section @sharbhar

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml [hunk 2]
```diff
@@ -178,72 +188,93 @@
...
-              '{{ app_apply_status.stdout }}' state."
-      when: not ("applied" in app_apply_status.stdout)
+          msg: "The apply of DM system app has failed.
+                Expected 'applied' state but the app is in
+                '{{ app_apply_status.stdout }}' state."
+        when: not ("applied" in app_apply_status.stdout)
+
+      when: get_software_version.stdout >= "24.09"
+
+    - block:
```
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:262]: In this block, please add the block to remove services and webhooks in: https://github.com/Wind-River/cloud-platform-deployment-manager/pull/326/files

**Review by yjian118 — CHANGES_REQUESTED** [review]
Need to add back the removal of the webhook stuff to support the reconfiguration in 22.12 below

**Review by hugonicodemos — APPROVED** [review]
lgtm

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml (hunk 2 above)
Comment (sharbhar) [reply docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:262]: Added

**Comment by sharbhar** [comment]
> Need to add back the removal of the webhook stuff to support the reconfiguration in 22.12 below

Added

**Review by yjian118 — APPROVED** [review]

**Pull request merged**